synapse/rest/client/room_batch.py

```diff
@@ -424,20 +425,26 @@ async def on_POST(
                 context=context,
             )
 
-        # Add the base_insertion_event to the bottom of the list we return
-        if base_insertion_event is not None:
-            event_ids.append(base_insertion_event.event_id)
+        insertion_event_id = event_ids.pop(0)
+        chunk_event_id = event_ids.pop(len(event_ids) - 1)
+        historical_event_ids = event_ids
```

Comment:
I'd find this more intuitive:

```suggestion
        insertion_event_id = event_ids[0]
        chunk_event_id = event_ids[-1]
        historical_event_ids = event_ids[1:-1]
```

Reply:
(for future reference, `event_ids.pop(len(event_ids) - 1)` can be written more efficiently as `event_ids.pop()`.)

Reply:
Good shout! I like this a lot better too to make it clear 😀